mesa/space.py

```diff
@@ -256,29 +259,88 @@ def get_neighborhood(
         if neighborhood is not None:
             return neighborhood
 
-        coordinates: set[Coordinate] = set()
-
         x, y = pos
-        for dy in range(-radius, radius + 1):
-            for dx in range(-radius, radius + 1):
-                # Skip coordinates that are outside manhattan distance
-                if not moore and abs(dx) + abs(dy) > radius:
-                    continue
 
-                coord = (x + dx, y + dy)
+        if moore:
+
+            x_from, x_to, x_max = x - radius, x + radius + 1, self.width
+            y_from, y_to, y_max = y - radius, y + radius + 1, self.height
+
+            if self.torus:
+
+                if radius >= x_max // 2:
```

Comment:
You should pack this slice calculation in a method/function like how it is done with `torus_adj`. And so it can be generalized to give the slice for both x and y.

It seems to me that the source of the speedup is due to the bound checking is only done once for all.

Reply:
in fact this way it's clearer

Reply:
now it should be all done

Reply:
yes anyway that is one of the reasons  plus that in the torus case it doesn't need to update the position and also that no sum between the deltas and the position is done inside the for loops